app/controllers/admin/address_controller.rb

```diff
@@ -0,0 +1,15 @@
+module Admin
+  class AddressController < AdminController
+    before_action :retreive_address_by_id, only: [:create, :edit, :update, :destroy]
+
+    private
+
+    def retreive_address_by_id
```

Comment:
these shouldn't be needed, you should be able to do this in the admin_controller. Sorry thats probably my fault for going over it both ways.